### ISSUE TITLE
timezone assignements causing multiple dates in streams with recent updates

### ISSUE THREAD

**Issue by scd** [theCrag/website#2088]
I have just noticed a structural issue in streams with multiple dates having recent information. The underlying issue is that two people can be updating different areas in the world at roughly the same time but, for example, the timezone offset means one update is assigned to 28th Sep (UTC +10) and the other 27th Sep (UTC -10).

Both updates may have been made within the last minute, however you have to scroll past all the older updates made on the 28th before you get to the updates made on the 27th.

This means there could be a systemic reason why some parts of the world rarely get to the top of the certain streams (eg UTC -10).

If you are just following local crags and people you know locally then it will not be so much of an issue, but if you are following everything in the whole world then it could be a problem.

It will be an issue for our world stream on the home page.

Not sure what to do about this.

**Comment by brendanheywood** [comment]
At first sight this looks similar to the other timezone issues but I don't think so. In the streams all we want to do is batch edits together, but the time of the stream event is just the last time in the batch. If we used the timezone of the crag it would be weird when I edited crags on the other side of the world, if we use my account timezone then we get the weird offset issue.

Instead we should view it purely as a batching mechanism, so instead of thinking of batching edited based on day, we batch edits based on how long it's been since the last edit in that crag. If it is less than X hours then append to the same batch and update the timestamp, if more then make a new batch. The timezone isn't in the equation at all, and the time actually rendered is in the watchers timezone.

We could (but I probably wouldn't) tune that X hours so that an editing session in the morning and another later in the evening on the same day end up as two stream events which does more closely match reality.
